independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java

```diff
@@ -867,12 +868,20 @@ public Object getFormParameter(String name, boolean single, boolean encoded) {
         }
         if (single) {
             FormData.FormValue val = formData.getFirst(name);
-            if (val == null || val.isFileItem()) {
+            if (val == null) {
                 return null;
             }
-            if (encoded) {
+
+            if (val.isFileItem()) {
```

Comment:
What case does this change cater to?

Reply:
When using:
```java
public class FormData {
    @FormParam("myFile")
    public String myFile;
}

```

Reply:
One idea is to force users to use `@PartType(MediaType.APPLICATION_OCTET_STREAM)` to copy the file content to a string. 
However, as far I understood, this is how resteasy classic works and what the spec says :/

Reply:
The spec doesn't say anything about `text` files though, does it? 
All I am questioning here is the need to support `String`. I would much rather not support it initially and have users manually convert the string if necessary.
Or we really want `String`, I think it only makes sense to do so for certain media types.

Reply:
I've just confirmed that Resteasy Classic works like this. 
If users try to load a binary file to String, I would say this is a user issue, but the framework should not stop this use case. 
Having said that and despite what Resteasy Classic does, I agree with you that we should force users to use the media type `@PartType(MediaType.APPLICATION_OCTET_STREAM)` if they want to copy the content of a file to a String. 

Let me know if you want me to do this.

Reply:
Sounds good

Reply:
PR updated